src/emulator/auth/operations.ts

```diff
@@ -114,6 +114,11 @@ function signUp(
     if (reqBody.idToken) {
       assert(!reqBody.localId, "UNEXPECTED_PARAMETER : User ID");
     }
+    if (reqBody.localId) {
+      // Fail fast if localId is taken (matching production behavior).
+      assert(!state.getUserByLocalId(reqBody.localId), "DUPLICATE_LOCAL_ID");
+    }
+
```

Comment:
Is this not tested?

Reply:
This is covered by integration tests in Node.js admin sdk.